cli/src/commands/extensions/permissions.rs

```diff
@@ -61,6 +62,74 @@ impl Permission {
             Err(anyhow!("Requires {resource_type} access to {resource:?}"))
         }
     }
+
+    pub fn subset_of(&self, parent: &Permission) -> Result<Permission> {
+        use Permission::*;
```

Comment:
I did this in order to avoid peppering the `match` statement with `Permission::`. I'd argue it is one of those use cases where _not_ having imports at the top helps legibility a great deal. Besides, it would not make sense to have this import at the top as the intent is just importing the _variants_ in scope.

Reply:
I **strongly** disagree with this improving legibility, it also very much goes against what is generally done in Rust.

The whole idea that the goal is "just importing the *variants* in scope" already shows why this is a terrible, terrible idea: You're basically intentionally creating an import that can be bound differently in a different scope, making things unnecessarily confusing. If an import is useful, it should be globally consistent in the module and enum variants generally should not be imported directly in Rust.